railties/lib/rails/engine/rake_proxy.rb

```diff
@@ -0,0 +1,34 @@
+require 'rake'
+require 'active_support'
+
+module Rails
+  class Engine
+    module RakeProxy #:nodoc:
+      private
+        def run_rake_task(command)
+          ARGV.unshift(command) # Prepend the command, so Rake knows how to run it.
+
+          Rake.application.standard_exception_handling do
+            Rake.application.init('rails')
+            Rake.application.load_rakefile
+            Rake.application.top_level
+          end
+        end
+
+        def rake_tasks
+          return @rake_tasks if defined?(@rake_tasks)
+
+          load_generators
+
+          Rake::TaskManager.record_task_metadata = true
+          Rake.application.init('rails')
+          Rake.application.load_rakefile
+          @rake_tasks = Rake.application.tasks.select(&:comment)
+        end
+
+        def formatted_rake_tasks
+          rake_tasks.map { |t| [ t.name_with_args, t.comment ] }
+        end
+    end
+  end
+end
```

Comment:
And I thought too much or reuse, but it is almost the same when the review. 
I'm sorry, and then modified to reuse the rake_proxy of Rails.